scipy/special/_support_alternative_backends.py

```diff
@@ -82,11 +79,13 @@ def __xlogy(x, y, *, xp=xp):
 
 
 def _chdtr(xp, spx):
+    if spx is None:
+        return None
     # The difference between this and just using `gammainc`
     # defined by `get_array_special_func` is that if `gammainc`
     # isn't found, we don't want to use the SciPy version; we'll
     # return None here and use the SciPy version of `chdtr`.
-    gammainc = getattr(spx, 'gammainc', None)  # noqa: F811
+    gammainc = getattr(spx.special, 'gammainc', None)  # noqa: F811
```

Comment:
So both `array_api_strict` and `CuPy` were ending up here, but *both* were not finding `gammainc` in `spx` because this should have been `spx.special`. Consequently, `None` was returned for both backends, and we fell back to NumPy for both backends. I think that's why we got CuPy errors in gh-20957.